backend/src/bundles/user-details/user-details.controller.ts

```diff
@@ -358,6 +370,51 @@ class UserDetailsController extends ControllerBase {
             payload: await this.userDetailsService.update(options.body),
         };
     }
+
+    private async searchUsers(
+        options: ApiHandlerOptions<{
+            query: UserDetailsSearchUsersRequestDto;
+        }>,
+    ): Promise<ApiHandlerResponse> {
+        const {
+            search,
+            jobTitle,
+            location,
+            isHired,
+            hardSkills,
+            experienceYears,
+            englishLevel,
+            employmentType,
+            BSABadges,
+        } = options.query;
+        const searchCriteria: UserDetailsSearchUsersRequestDto = {
+            ...(search !== undefined && { search }),
+            ...(jobTitle !== undefined && { jobTitle }),
+            ...(location !== undefined && { location }),
+            ...(isHired !== undefined && { isHired }),
+            ...(hardSkills !== undefined && { hardSkills }),
+            ...(experienceYears !== undefined && { experienceYears }),
+            ...(englishLevel !== undefined && { englishLevel }),
+            ...(employmentType !== undefined && { employmentType }),
+            ...(BSABadges !== undefined && { BSABadges }),
+        };
+
+        if (Object.keys(searchCriteria).length === 0) {
+            throw new HttpError({
+                status: HttpCode.BAD_REQUEST,
+                message: ErrorMessages.USER_NOT_FOUND,
+            });
+        }
```

Comment:
Should I implement the findAll method and use it if the query is empty?

Reply:
Or just leave like that, it returns all users with empty query params anyway

Reply:
You should return all users if query is empty, because if we sent empty filters we expect that we get all users without filter, only if we provide some filters we need to filter